src/transformers/generation/flax_utils.py

```diff
@@ -581,6 +587,22 @@ def _greedy_search(
         # per batch-item holding current token in loop.
         sequences = jnp.full((batch_size, max_length), pad_token_id, dtype=jnp.int32)
         sequences = lax.dynamic_update_slice(sequences, input_ids, (0, 0))
+        if hasattr(self.config, "vocab_size") and self.config.vocab_size is not None:
```

Comment:
Can we guarantee every model compatible with generate has `"vocab_size"` in its config?

Reply:
I tried to support all models that has `vocab_size` and pass all test codes due to, but there may or will be some models which does not have `vocab_size`. 
So I've added a TypeError raised when `output_scores` is set to True when calling `generate()` while there is no recognized `vocab_size` in its configs or in the decoder or encoder of it.
I've also return `scores=None` when the `output_scores` is set to False.
Please refer to the commit: https://github.com/huggingface/transformers/pull/22700/commits/dddc0fcd1ae4f4d9796e18acb5bef45b2b15a1e6